app/src/main/java/org/mozilla/fenix/compose/list/ExpandableListHeader.kt

```diff
@@ -45,16 +45,21 @@ fun ExpandableListHeader(
     expanded: Boolean? = null,
     expandActionContentDescription: String? = null,
     collapseActionContentDescription: String? = null,
-    onClick: () -> Unit = {},
+    onClick: (() -> Unit)? = null,
```

Comment:
These changes were needed to be able to handle clicks in the parent and in so show the expected indication.
Otherwise, the indication (and clicks) would look like

https://user-images.githubusercontent.com/11428869/160660920-d6c535b8-6db0-45fd-9a18-4081fc2bc400.mp4

Reply:
that's a weird interaction. thanks for the fix. this will also make it so any headers without click lambdas won't show the click ripple.

Reply:
Extracted this PR out in https://github.com/mozilla-mobile/fenix/pull/24898